### PR TITLE
Sharepoint support

### PR DESCRIPTION
**Pull request by jacobtoppm**
- Adds new DocxParser:
    - Includes text, bold, italics, and headings
- Adds new MicrosoftPicker for import from OneDrive/SharePoint
- Restyles button to read "import" and open menu on click to show sources

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,9 +20,9 @@
 
 setup(
     name='wagtail_content_import',
-    version="0.1.6",
+    version="0.2.0",
     description='A module for Wagtail that provides functionality for importing page content from third-party sources.',
-    author='Samir Shah, Jacob Topp-Mugglestone, Karl Hobley, Matthew Westcott',
+    author='Samir Shah, Jacob Topp-Mugglestone,  Karl Hobley, Matthew Westcott',
```
Comment (tomdyson) [inline setup.py]: extra space

File: wagtail_content_import/parsers/__init__.py [hunk 2]
```diff
@@ -15,4 +15,18 @@ def get_google_parser():
...
     parser_string = get_google_parser_string()
 
-    return import_string(parser_string)
\ No newline at end of file
+    return import_string(parser_string)
+
+
+def get_docx_parser_string():
+    """
+    Get the dotted ``app.Model`` name for the Google Docs parser as a string.
```
Comment (tomdyson) [inline wagtail_content_import/parsers/__init__.py]: "Google Docs parser" - should this be "Office Open XML parser"?

File: wagtail_content_import/pickers/microsoft/static/wagtail_content_import/microsoft_picker.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    class MicrosoftPicker {
+        constructor(redirectUri, clientId, createPageUrl, csrfToken) {
+            this.redirectUri = redirectUri;
+            this.clientId = clientId;
+            this.createPageUrl = createPageUrl;
+            this.csrfToken = csrfToken;
+        }
+
+        post_url(response) {
+                                // POST the download url of the document to the create page view
```
Comment (tomdyson) [inline wagtail_content_import/pickers/microsoft/static/wagtail_content_import/microsoft_picker.js]: massive indent :)

File: wagtail_content_import/pickers/microsoft/utils.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def parse_document(document):
+    parser = get_docx_parser()
+    return parser(document).parse()
+
+
+class MicrosoftPicker(Picker):
+    name = "microsoft"
+    verbose_name = "SharePoint"
```
Comment (tomdyson) [inline wagtail_content_import/pickers/microsoft/utils.py]: is "SharePoint" accurate here? AFAIK the same picker works for OneDrive and SharePoint.

File: setup.py (hunk 1 above)
Comment (jacobtoppm) [reply setup.py]: Ah sorry - fixed just after committing, and recommitted!

File: wagtail_content_import/pickers/microsoft/utils.py (hunk 4 above)
Comment (jacobtoppm) [reply wagtail_content_import/pickers/microsoft/utils.py]: Hmm - this is only used in the menu button text. I'm not sure which is better, OneDrive/SharePoint is quite long, but you're right, it works for both.

Microsoft, maybe? Or OneDrive as it is more commonly used?

File: wagtail_content_import/parsers/__init__.py (hunk 2 above)
Comment (jacobtoppm) [reply wagtail_content_import/parsers/__init__.py]: Thanks for catching that - I've corrected now.

File: wagtail_content_import/pickers/microsoft/static/wagtail_content_import/microsoft_picker.js (hunk 3 above)
Comment (jacobtoppm) [reply wagtail_content_import/pickers/microsoft/static/wagtail_content_import/microsoft_picker.js]: ...Oops. Corrected now!

File: wagtail_content_import/pickers/microsoft/utils.py (hunk 4 above)
Comment (tomdyson) [reply wagtail_content_import/pickers/microsoft/utils.py]: I vote OneDrive

Comment (jacobtoppm) [reply wagtail_content_import/pickers/microsoft/utils.py]: Great, thanks, I've changed it over in latest commit.

File: wagtail_content_import/pickers/microsoft/static/wagtail_content_import/microsoft_picker.js [hunk 5]
```diff
@@ -8,27 +8,27 @@
...
-                                microsoftDocField.name = 'microsoft-doc';
-                                microsoftDocField.value = response.value[0]["@microsoft.graph.downloadUrl"];
-                                form.appendChild(microsoftDocField);
+                             let microsoftDocField = document.createElement('input');
+                             microsoftDocField.type = 'hidden';
+                             microsoftDocField.name = 'microsoft-doc';
+                             microsoftDocField.value = response.value[0]["@microsoft.graph.downloadUrl"];
+                             form.appendChild(microsoftDocField);
 
-                                form.submit();
```
Comment (zerolab) [inline wagtail_content_import/pickers/microsoft/static/wagtail_content_import/microsoft_picker.js]: It is still very indented. https://github.com/torchbox/wagtail-content-import/blob/7087065b21a5a7968ec27903e21d1024f3e725d4/wagtail_content_import/pickers/microsoft/static/wagtail_content_import/microsoft_picker.js

Comment (jacobtoppm) [reply wagtail_content_import/pickers/microsoft/static/wagtail_content_import/microsoft_picker.js]: Ah, sorry. I think I've fixed it now.

**Pull request merged**